Add team filter for runs

iomodo opened the pull request:
#### Summary
This PR
- Removes team name from the title
- Displays runs for all the teams by default
- Appends team name to the run name in the table
- Filters runs by team
- Does not display team name or team filter at all if user is in the single team.
- updates runs url to `plug/com.mattermost.plugin-incident-management/runs` (no team)


https://user-images.githubusercontent.com/2340127/124115282-ab52d900-da7e-11eb-8bd4-3d6dadd7d1aa.mp4


#### Ticket Link
  Fixes: https://github.com/mattermost/mattermost-server/issues/XXXXX


#### Checklist
<!-- Check off items as they are completed. ~~Strike through~~ items if they don't apply -->
- [ ] Telemetry updated
- [ ] Gated by experimental feature flag
- [x] Unit tests updated

agarciamontoro commented on webapp/src/components/team/team_with_icon.tsx:
```diff
@@ -0,0 +1,39 @@
+
+interface Props {
+    team: Team;
+    classNames?: Record<string, boolean>;
+    className?: string;
+    extra?: JSX.Element;
+    withoutIcon?: boolean;
+    withoutName?: boolean;
```
I have not seen these props being used anywhere. Should we get rid of them?

agarciamontoro commented on webapp/src/components/team/team_icon.scss:
```diff
@@ -0,0 +1,105 @@
+.TeamIcon__content {
```
Same styled-component vs CSS comment.

agarciamontoro commented on webapp/src/components/backstage/playbook_runs/playbook_run_list/playbook_run_list.tsx:
```diff
@@ -50,20 +52,28 @@ import {startPlaybookRun} from 'src/actions';
 );
 
+const OwnerControlComponent = (ownProps: ControlProps<any>) => {
+    return controlComponent(ownProps, 'Owners');
```
We're changing this to uppercase, is this by design? (same for the `TeamControlComponent`)

iomodo replied:
nope, just a mistake, changing back to the lowercase

agarciamontoro commented on webapp/src/components/backstage/playbook_runs/playbook_run_list/playbook_run_list.tsx:
```diff
@@ -187,18 +197,16 @@ const BackstagePlaybookRunList = () => {
+        setFetchParams((oldParams) => ({...oldParams, status: queryForTeamID || ''}));
+    }, [query]);
 
     // When the component is first mounted (or the team changes), determine if there are any
```
We should update this comment
```suggestion
    // When the component is first mounted, determine if there are any
```

agarciamontoro commented on webapp/src/components/team/team_selector.tsx:
```diff
@@ -0,0 +1,263 @@
+}
+
+const ChildContainer = styled.div<ChildContainerProps>`
+    top: ${(props) => 27 - (props.moveUp || 0)}px;
```
Nit: should we move this 27 to a constant?

agarciamontoro reviewed (changes requested):
Great work, thank you! :tada: I love seeing that we are going team-agnostic <3

I left a couple of comments below, and have a few general considerations:
1. I found the TeamIcon, TeamWithIcon and TeamSelector too flexible. Do we need all that flexibility or can we simplify the code there?
2. Why are we using a feature branch here? I guess there is no possibility of using the experimental feature flag here, right?

itao commented:
I noticed the emphasis on `/plug/` in the URL, is it necessary to have it there as opposed to removing completely?

Functionally the UX seems to be all there 👍  there are some quirks like the avatar for owner/team looks a bit too large next to the text and the `reset to all team` is a different pattern from other places such as timeline filter (see screenshot below). Looking to @abhijit-singh for more thorough UI review

![Screen Shot 2021-07-06 at 9 02 56 AM](https://user-images.githubusercontent.com/725414/124604571-0f87eb00-de39-11eb-9580-4eee163a3c84.png)

iomodo commented on server/sqlstore/playbook_run.go:
```diff
@@ -801,6 +803,20 @@ func (s *playbookRunStore) buildPermissionsExpr(info app.RequesterInfo) sq.Sqliz
+		return sq.Eq{"i.TeamID": teamID}
+	}
+
+	return sq.Expr(`
+		EXISTS(SELECT 1
+					FROM TeamMembers as tm
+					WHERE tm.TeamId = i.TeamID
+					  AND tm.DeleteAt = 0  
+		  	  		  AND tm.UserId = ?)
+		`, userID)
+}
```
This filters out all the playbooks that belong to the team where user is not a part of.

agarciamontoro replied:
Ooh, that's right. I don't know how I read this last time :shrug:

iomodo commented:
Addressed UI changes @abhijit-singh suggested. 
Addressed @agarciamontoro's suggestions.
Added all the css vars we use for the theme styling
Adding @crspeller for the review


https://user-images.githubusercontent.com/2340127/125069282-9b16ab80-e0c7-11eb-915c-4f044b211ccf.mp4

agarciamontoro commented on webapp/src/components/backstage/css_utils.tsx:
```diff
@@ -0,0 +1,49 @@
+} from 'mattermost-redux/utils/theme_utils';
+import cssVars from 'css-vars-ponyfill';
+
+export function applyTheme(theme: any) {
```
Oh, cool! I'm slightly concerned this could get out of sync with the corresponding code in the webapp, though. Is there any ideal world where we can import it from there?

agarciamontoro reviewed (approved):
Looks great! Thank you for addressing all my nit comments :)

abhijit-singh reviewed (approved):
Looks awesome @iomodo! Thank you!

Just one small nit, there seems to be an on-click state with a blue background in the team filter menu and the button. Haven't seen that anywhere else in the product, can we get rid of that?
![image](https://user-images.githubusercontent.com/14369607/125306341-bf36ee80-e34c-11eb-989d-d46231c4bcc0.png)

iomodo commented:
@crspeller gentle ping to review theme solution for a custom route

crspeller reviewed (dismissed):
The apply theme thing is a good temporary solution. The global header will solve this in general.

iomodo commented:
After some discussions with @crspeller 
1. Removed apply theme and routing from this PR
2. Rebased to master
3. Fixed unit tests(in fact removed them, because they where no longer valid).
4. Removed the onClick state pointed by @abhijit-singh

crspeller commented on tests-e2e/cypress/integration/backstage/playbook_run_list_spec.js:
```diff
@@ -78,65 +78,65 @@ describe('backstage playbook run list', () => {
-            playbookId,
-            playbookRunName,
-            ownerUserId: userId,
+    /*
```
Could we remove them instead of commenting them out if they are no longer valid?
